trin-core/src/utp/stream.rs

```diff
@@ -1097,6 +1113,25 @@ impl UtpStream {
                 }
 
                 if let Some(last_seq_nr) = self.send_window.last().map(Packet::seq_nr) {
+                    // Remove all acknowledged packets from the send window
+                    let ack_packets = extension
+                        .iter()
+                        .enumerate()
+                        .filter(|&(_, received)| received)
+                        .map(|(idx, _)| packet.ack_nr() + 2 + idx as u16)
+                        .take_while(|&seq_nr| seq_nr <= last_seq_nr);
+
+                    for seq_nr in ack_packets {
+                        if let Some(position) = self
+                            .send_window
+                            .iter()
+                            .position(|packet| packet.seq_nr() == seq_nr)
+                        {
+                            let packet = self.send_window.remove(position);
+                            self.cur_window = self.cur_window.saturating_sub(packet.len() as u32);
```

Comment:
Should we be panicking here instead of `saturating_sub`? If we overflow, does that reflect some incorrect accounting on our part, or is this defense from some incorrect behavior from the peer?

Reply:
Thinking about it now, panicking makes more sense because it will indicate a bug in our code.